src/chopsticks.rs

```diff
@@ -0,0 +1,679 @@
+	let http_endpoint = format!("http://127.0.0.1:{}", network_config.port);
+
+	// Extract call data for injections based on the actual HackMD flow
+	let (preimage_call_data, whitelist_call_data, dispatch_call_hash, dispatch_call_len) =
+		extract_flow_data(calls);
+
+	format!(
+		r#"
+/**
+ * Simple HTTP-based chopsticks interaction function
+ */
+async function rpcCall(method, params = []) {{
+	const http = require('http');
+	
+	const postData = JSON.stringify({{
+		id: Math.floor(Math.random() * 1000),
+		jsonrpc: '2.0',
+		method,
+		params
+	}});
+	
+	return new Promise((resolve, reject) => {{
+		const req = http.request({{
+			hostname: '127.0.0.1',
+			port: 8000,
+			path: '/',
+			method: 'POST',
+			headers: {{
+				'Content-Type': 'application/json',
+				'Content-Length': Buffer.byteLength(postData)
+			}}
+		}}, (res) => {{
+			let data = '';
+			
+			res.on('data', (chunk) => {{
+				data += chunk;
+			}});
+			
+			res.on('end', () => {{
+				try {{
+					const result = JSON.parse(data);
+					if (result.error) {{
+						reject(new Error(`RPC error: ${{result.error.message}}`));
+					}} else {{
+						resolve(result.result);
+					}}
+				}} catch (error) {{
+					reject(new Error(`Failed to parse response: ${{error.message}}`));
+				}}
+			}});
+		}});
+		
+		req.on('error', (error) => {{
+			reject(new Error(`HTTP request failed: ${{error.message}}`));
+		}});
+		
+		req.write(postData);
+		req.end();
+	}});
+}}
+
+async function main() {{
+	console.log('🔗 Testing chopsticks connectivity at {}...');
+	
+	try {{
+		// Test basic connectivity
+		await testChopsticksConnection('{}');
+		
+		console.log('📤 Simulating preimage submission...');
+		console.log('Preimage call data: {}');
+		
+		console.log('🏛️ Injecting fellowship whitelist call...');
+		await injectFellowshipCall('{}', '{}');
+		
+		console.log('📊 Injecting whitelisted caller dispatch...');
+		await injectWhitelistedCallerCall('{}', {}, '{}');
+		
+		console.log('🧪 Running user-defined tests...');
+		{}
+		
+		console.log('✅ All chopsticks tests completed successfully!');
+	}} catch (error) {{
+		console.error('❌ Test failed:', error.message);
+		process.exit(1);
+	}}
+}}
+
+async function testChopsticksConnection(endpoint) {{
+	try {{
+		const result = await rpcCall('system_health');
+		console.log('✅ Chopsticks is running and responsive');
+		console.log('Health check result:', result);
+	}} catch (error) {{
+		throw new Error(`Chopsticks not responding: ${{error.message}}`);
+	}}
+}}
+
+async function injectFellowshipCall(endpoint, callData) {{
+	// Get current block number
+	const header = await rpcCall('chain_getHeader');
+	const currentBlock = parseInt(header.number, 16);
+	const targetBlock = currentBlock + 1;
+	
+	console.log(`Current block: ${{currentBlock}}, injecting for block: ${{targetBlock}}`);
+	
+	// Inject fellowship call into scheduler
+	await rpcCall('dev_setStorage', [{{
+		scheduler: {{
+			agenda: [
```

Comment:
Great, I see you already know how to do this!

This isn't immediately relevant, but post-AHM, the manner in which you choose the target block will change.

Reply:
Thanks for pointing that out! Is there anything you think I should adjust in the current implementation to better prepare for post-AHM changes?